Artiom/src/main/java/DateTemplate.java

```diff
@@ -0,0 +1,9 @@
+public class DateTemplate {
+
+    public static void main(String[] args) {
+        String dateRegex = "2020-09-03";
```

Comment:
Looks fine but lets make it pass the following tests as well:
0000-00-00 - is not a valid date
0000-19-00 - is not a valid date
0000-00-39 - is not a valid date
2020a03b08 - is not a valid date

Reply:
done

Reply:
Your version is still not cover above test cases.
As an example you can try the next one regex:

^([1-9]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01]))$

year:
[1-9] - can match the number 1 to 9
\d{3} - searches for three digits

month:
0[1-9] - starts from 0 and following digit between 1 and 9 (month from 01 to 09)  
| - OR 
1[0-2] - starts from 1 and following digit between 0 and 2 (month: 10, 11, 12)

day: 
0[1-9] - starts from 0 and following digit between 1 and 9
|- OR 
[12]\\d - digit 1 or 2 with following any digit (\\d - short for [0-9])
|- OR 
3[01] -  starts from 3 and following digit 0 or 1

Reply:
done